model/EligibilityService.php

```diff
@@ -548,4 +590,20 @@ private function createExecutionContext($deliveryExecution, $testCenter)
 
         return $executionContext;
     }
+
+    /**
+     * @throws \common_exception_InconsistentData|\core_kernel_persistence_Exception
+     */
+    public function deleteEligibilitiesWithoutDelivery()
```

Comment:
This method should be also moved to separate file. Probably it would be better to add Report in Updater script with recommendation to run it manually  and do not execute it directly, for example like here - fix/TAO-7778/delete-elegibility-afret-deleting-delivery.
It may be really time consuming on bigger amount of data, therefore it's not recommended to run it from Updater.

Reply:
oat\taoTestCenter\scripts\tools\CleanupEligibility modifited (add checking for delivery existing)
and added warning to update script